src/trading.rs

```diff
@@ -298,6 +298,34 @@ impl Tarkov {
         handle_error(res.error, res.data)
     }
 
+    /// Get list of price trader is willing to pay for in users inventory
+    pub async fn get_trader_prices(&self, trader_id: &str) -> Result<HashMap<String, Price>> {
+        if trader_id.is_empty() {
+            return Err(Error::InvalidParameters);
+        }
+
+        let mut result: HashMap<String, Price> = HashMap::new();
+
+        let mut prices = self.get_trader_prices_raw(trader_id).await?;
+
+        for (id, price_w) in prices.drain() {
+            if price_w.len() != 1 {
+                panic!("The price structure top is wrong");
```

Comment:
We shouldn't panic but instead return a proper error type.

Reply:
Yeah, I originally had a proper error type here, but I saw that in places like the `get_trader_prices_raw` if the response doesn't match up, you panic with `expect`. I am all for errors though, I just couldn't find an appropriate error type.